### PR TITLE
Merging purgatory into osm_app branch

### PR DESCRIPTION
**Pull request by nshoemaker**
Because I cannot commit to the g3m repo myself, I am making a pull request so @boyP can pull my changes into our branch.

All I have done so far is merge the changes from purgatory into the osm_app branch. Also checked that the project builds correctly.

### CODE REVIEW

**Comment by nshoemaker** [comment]
I have also started creating BoxShapes for each building. Currently testing on the first 10 buildings of each tile. Currently, the BoxShapes seem to be located at places other than where I would want them to be. Not sure if I am feeding position in wrong, or if things are growing upside down, through the earth, and back out the other side. Still need to debug.

File: iOS/G3MApp/G3MApp/G3MOSMBuildingsDemoScene.cpp [hunk 1]
```diff
@@ -18,9 +18,16 @@
 #include <G3MiOSSDK/JSONArray.hpp>
 #include <G3MiOSSDK/JSONNumber.hpp>
 #include <G3MiOSSDK/Mark.hpp>
+#include <G3MiOSSDK/Box.hpp>
```
Comment (nshoemaker) [inline iOS/G3MApp/G3MApp/G3MOSMBuildingsDemoScene.cpp:21]: Get rid of this we are no longer using the Box class.

**Comment by DiegoGomezDeck** [comment]
> Currently, the BoxShapes seem to be located at places other than where I would want them to be.

Take in mind the "zero" of the Box's coordinate-system is its center.

**Comment by nshoemaker** [comment]
Thanks, @DiegoGomezDeck! We realized this and moved our position coordinates to the center of the box. 

Originally, we were actually running into undefined behavior due to pointer issues with the position coordinates we gave the BoxShape. 

We just fixed both problems though and so far, so good.

**Comment by boyP** [comment]
ios simulator screen shot apr 11 2015 3 58 55 pm

**Pull request merged**